vllm/attention/backends/flash_attn.py

```diff
@@ -18,6 +18,135 @@
 if TYPE_CHECKING:
     from vllm.worker.model_runner import ModelInputForGPUBuilder
 
+torch.library.define("vllm::flash_attn_varlen_func",
```

Comment:
I believe `torch.library.define` is generally not necessary. We should use high-level APIs listed in https://pytorch.org/docs/main/library.html .

Reply:
I tried using `custom_op` here but it couldn't deal with the window_size tuple so I ended up sticking with `define`

Reply:
I think the longer term answer is to move the registration into the flash attn module itself once it is pulled into the main vllm build process.